src/test/scala/sigmastate/serialization/TypeSerializerSpecification.scala

```diff
@@ -84,4 +85,9 @@ class TypeSerializerSpecification extends SerializationSpecification {
     roundtrip(STuple(SLong, SLong, SByte, SBoolean, SInt),
       Array[Byte](TupleTypeCode, 5, SLong.typeCode, SLong.typeCode, SByte.typeCode, SBoolean.typeCode, SInt.typeCode))
   }
+
+  property("tuple of tuples crazy deep") {
```

Comment:
@ergomorphic it's throwing `StackOverflow` now. We should have some depth limit.

Reply:
yes, that is what I mean by malicious tests. It is better to have synthetic tests which deterministically reporduce specific cases, like very deep trees or invalid codes etc. 
They should cover all existing error throwing cases.

Reply:
@greenhat We need to have something similar to Corpus described here http://www.llvm.org/docs/LibFuzzer.html

Reply:
@ergomorphic I've successfully tested the proof-of-concept for coverage-guided fuzzying in scala (based on scoverage). I've started the project https://github.com/greenhat/scalafuzz-scalac-plugin and I plan to have a random mutations working this week.